src/components/utils/src/threads/thread_posix.cc

```diff
@@ -100,11 +102,14 @@ void* Thread::threadFunc(void* arg) {
   pthread_cleanup_push(&cleanup, thread);
 
   thread->state_lock_.Acquire();
+  thread->thread_created_ = true;
```

Comment:
@orhan-mehmedov, 
Do we need to update comment at the beginning of this function?